apps/native/app/src/screens/passkey/passkey.tsx

```diff
@@ -0,0 +1,229 @@
+        style={{ marginHorizontal: 16 }}
+      />
+      <SafeAreaView style={{ flex: 1 }}>
+        <View
+          style={{
+            justifyContent: 'center',
+            alignItems: 'center',
+            flex: 1,
+          }}
+        >
+          <Image
+            source={logo}
+            resizeMode="contain"
+            style={{ width: 45, height: 45 }}
+          />
+          <Text>
+            <Typography
+              variant={'heading2'}
+              style={{
+                paddingHorizontal: theme.spacing[2],
+                marginBottom: theme.spacing[2],
+              }}
+              textAlign="center"
+            >
+              <FormattedMessage
+                id="passkeys.headingTitle"
+                defaultMessage="Innskrá með Ísland.is appinu"
+              />
+            </Typography>
+            <Typography textAlign="center">
+              <FormattedMessage
+                id={
+                  url
+                    ? 'passkeys.openUrlHeadingSubtitle'
+                    : 'passkeys.headingSubtitle'
+                }
+                defaultMessage={
+                  url
+                    ? 'Þú ert að fara að opna Ísland.is í vafra. Viltu búa til aðgangslykil til að skrá þig inn sjálfkrafa með appinu?'
+                    : 'Viltu búa til aðgangslykil til að skrá þig inn sjálfkrafa með appinu?'
+                }
+              />
+            </Typography>
+          </Text>
+          <Image
+            source={illustrationSrc}
+            style={{ width: 195, height: 223 }}
+            resizeMode="contain"
+          />
+        </View>
+        <View
+          style={{
+            paddingHorizontal: theme.spacing[2],
+            paddingVertical: theme.spacing[4],
+          }}
+        >
+          <Button
+            title={intl.formatMessage({
+              id: 'passkeys.createButton',
+              defaultMessage: 'Búa til aðgangslykil',
+            })}
+            onPress={async () => {
+              try {
+                setIsLoading(true)
+                // Don't show lockscreen behind native passkey modals
+                authStore.setState(() => ({
+                  noLockScreenUntilNextAppStateActive: true,
+                }))
+
+                const registered = await registerPasskey()
+
+                if (!registered) {
+                  setIsLoading(false)
+                }
+
+                // If we don't get url we are only registering, close modal after registering
+                if (registered && !url) {
+                  setIsLoading(false)
+                  Navigation.dismissModal(componentId)
+                }
+
+                if (registered && url) {
+                  // Don't show lockscreen behind native passkey modals
+                  authStore.setState(() => ({
+                    noLockScreenUntilNextAppStateActive: true,
+                  }))
+
+                  const passkey = await authenticatePasskey()
```

Comment:
~we're creating an authentication with a stored passkey, so variable should be `authentication` or `authenticationResponse`?